trin-state/src/lib.rs

```diff
@@ -51,13 +62,7 @@ pub async fn main() -> Result<(), Box<dyn std::error::Error>> {
     tokio::spawn(async move {
         let (mut p2p, events) = PortalnetProtocol::new(portalnet_config).await.unwrap();
```

Comment:
Inside `PortalnetProtocol` it's launching discv5 right now. So currently, it would be launching discovery 3 different times. The direction I would head is to start discovery with trin-core, and pass a handle into both `trin-state` and `trin-history` so that neither of them try to start up their own discovery instance.

Reply:
Hmm, my understanding might be wrong here? But this is in the "main()" function which will only be called if somebody chooses to run this binary by itself (`cargo run -p trin-state`). In which case, wouldn't it want to boot up it's own base discv5? I guess maybe there's some more thinking to be done about the "solo binary" story, do we want to support it, and be a bit more explicit about the functionality it would support.

Reply:
When booting trin via `cargo run`, inside `trin/src/lib.rs` we only call `initialize()` for each subnet, not their `main()`.

Reply:
Ah yes, I had missed that. But... then aren't there other parts of the `PortalnetProtocol` class that should be run that aren't being run? Like creating `PortalnetEvents` and running `tokio::spawn(events.process_discv5_requests());`. That's the code that reacts to the inbound talkreq's, and serves the overlay protocol to peers.

Reply:
Yeah, for sure those are needed - but to keep things simple, I just ignored it for this pr 😅  - tracking it [here](https://github.com/ethereum/trin/issues/93)